core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -489,10 +495,6 @@ public void loadNext() throws IOException {
           }
           upstream = nextUpstream;
           nextUpstream = null;
-
-          assert(inMemSorter != null);
```

Comment:
What happens if we don't spill? Will we still free the sorter? I'm wondering whether we should also try to free it here in that case, maybe by changing assert into `if`.